express-error-handling/server/index.js

```diff
@@ -0,0 +1,223 @@
+require('dotenv/config');
+const pg = require('pg');
+const express = require('express');
+const ClientError = require('./client-error'); // eslint-disable-line
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.get('/api/grades', (req, res, next) => {
+  const sql = `
+    select *
+      from "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      const grades = result.rows;
+      res.json(grades);
+    });
+  throw new ClientError(400, 'an unexpected error occurred')
+    .catch(err => next(err));
+  // .catch(err => {
+  //   // console.error(err);
+  //   // res.status(500).json({
+  //   //   error: 'an unexpected error occurred'
+  //   // });
+  // });
+});
+
+app.post('/api/grades', (req, res, next) => {
+  const { name, course } = req.body;
+  const score = Number(req.body.score);
+  if (!Number.isInteger(score) || score < 0 || score > 100) {
+    throw new ClientError(400, 'score must be an integer between 0 and 100')
+      .catch(err => next(err));
```

Comment:
The `catch` only gets added to a function that returns a promise, like `db.query`. So no need for `catch` here